examples/private_set_intersection/client/rust/src/main.rs

```diff
@@ -0,0 +1,187 @@
+    let public_key_tag = web_assembly_module_signature_tag(&public_key);
+    let mut label = Vec::new();
+    Label {
+        confidentiality_tags: vec![public_key_tag],
+        integrity_tags: vec![],
+    }
+    .encode(&mut label)
+    .context("Error encoding label")?;
+    Ok(PrivateSetIntersectionClient::with_interceptor(
+        channel,
+        move |mut request: Request<()>| {
+            request.metadata_mut().insert_bin(
+                oak_abi::OAK_LABEL_GRPC_METADATA_KEY,
+                MetadataValue::from_bytes(label.as_ref()),
+            );
+            Ok(request)
+        },
+    ))
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::init();
+    let opt = Opt::from_args();
+
+    let uri = opt.uri.parse().context("Error parsing URI")?;
+    let root_tls_certificate = tokio::fs::read(&opt.root_tls_certificate)
+        .await
+        .context("Couldn't load certificate file")?;
+    let public_key_file = tokio::fs::read(&opt.public_key)
+        .await
+        .context("Couldn't load public key file")?;
+    let public_key_bytes = pem::parse(public_key_file)
+        .context("Empty public key file")?
+        .contents;
+
+    // Submit sets from different clients.
+    let channel_1 = create_channel(&uri, &root_tls_certificate)
+        .await
+        .context("Couldn't create channel")?;
+    let mut client_1 = create_client(channel_1, &public_key_bytes)
+        .await
+        .context("Couldn't create gRPC client")?;
+    let request = Request::new(SubmitSetRequest {
+        values: vec!["a".to_string(), "b".to_string(), "c".to_string()],
+    });
+    client_1
+        .submit_set(request)
+        .await
+        .context("Couldn't submit set")?;
+
+    let channel_2 = create_channel(&uri, &root_tls_certificate)
+        .await
+        .context("Couldn't create channel")?;
+    let mut client_2 = create_client(channel_2, &public_key_bytes)
+        .await
+        .context("Couldn't create gRPC client")?;
+    let request = Request::new(SubmitSetRequest {
+        values: vec!["b".to_string(), "c".to_string(), "d".to_string()],
+    });
+    client_2
+        .submit_set(request)
+        .await
+        .context("Couldn't submit set")?;
+
+    // Use an invalid public key.
+    let invalid_public_key_bytes = base64::decode("vpxqTZOUq1FjcaB9uJYCuv4kAg+AsgMwubA6WE+2pmk=")
+        .context("Couldn't decode public key")?;
+    let channel_3 = create_channel(&uri, &root_tls_certificate)
+        .await
+        .context("Couldn't create channel")?;
+    let mut invalid_client = create_client(channel_3, &invalid_public_key_bytes)
+        .await
+        .context("Couldn't create gRPC client")?;
+
+    let request = Request::new(SubmitSetRequest {
+        values: vec!["c".to_string(), "d".to_string(), "e".to_string()],
+    });
+    invalid_client
+        .submit_set(request)
+        .await
+        .err()
+        .context("Submitted set with an invalid public key label")?;
```

Comment:
I am not sure I am reading this correctly, I thought you would expect an error in this case? Could you make it clearer anyways?

Reply:
Yes, it fails if there is no error. `.err()` returns `Some` if there is error and `None` otherwise.
Added a comment